File: 11-20/euler_19.py
import time
startTime = time.clock()
months = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
#1 is a monday, 7 is a sunday, etc.
weekday = 1
year = 1901
month = 1
monthDay = 1
total = 0
while year < 2001:
    if weekday == 7 and monthDay == 1:
        total += 1
        print("day", weekday, "monthDay", monthDay, "month", month, "year", year)
    if year % 4 == 0:
        months[1] = 29
    elif not months[1] == 28:
        months[1] = 28
    if weekday == 7:
        weekday = 1
    else:
        weekday += 1
    if monthDay >= months[month - 1]:
        month += 1
        monthDay = 1
    else:
        monthDay += 1
    if month > 12:
        month = 1
        year += 1
print("--- answer: %s ---" % total)
print("--- %s seconds ---" % (time.clock() - startTime))

File: Euler_C/euler_19.c
#include <stdio.h>
#include <time.h>

int main(){
  clock_t start = clock();
  const int days[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
  int dayOfWeek = 1; //1 is monday, 2 is tuesday, 7 is sunday
  int total = 0;
  int leap;
  for(int year = 1900; year <= 2000; year++){
    int leap = (year % 4 == 0 && (year % 400 == 0 || !(year % 100 == 0)));
    for(int month = 1; month <= 12; month++){
      if(dayOfWeek == 7 && year > 1900){
        total++;
      }
      for(int day = 1; day <= days[month - 1] + (month == 2 && leap); day++){
        dayOfWeek = dayOfWeek == 7 ? 1 : dayOfWeek + 1; //adds one, otherwise loops around
      }
    }
  }
  printf("Amount of Sundays on the first is %ld\n", total);
  printf("Executed in: %.6f seconds.\n", (double)(clock() - start) / CLOCKS_PER_SEC);
}
